list.go

```diff
@@ -253,16 +252,14 @@ func listReferrers(writer io.Writer, opts listOptions) error {
 	}
 
 	remoteOpts := append(opts.RemoteOptions(), remote.WithAuthFromKeychain(authn.DefaultKeychain))
-	index, err := remote.Referrers(targetDigest, remoteOpts...)
+	imageIndex, err := remote.Referrers(targetDigest, remoteOpts...)
+	if err != nil {
+		return fmt.Errorf("error fetching referrers: %w", err)
+	}
 
+	index, err := imageIndex.IndexManifest()
 	if err != nil {
-		if e, ok := err.(*transport.Error); ok && e.StatusCode == 404 {
-			// If the OCI registry returns 404, process it as an index with no referrer. This happens when the OCI registry does not support
-			// the referrers API.
-			index = &v1.IndexManifest{}
```

Comment:
This code was originally implemented to avoid a past bug, and is no longer necessary.
The original API now returns an empty IndexManifest as well.

This issue has been fixed in this [PR](https://github.com/google/go-containerregistry/pull/1648).